src/parser/core/modules/PrecastStatus.ts

```diff
@@ -0,0 +1,153 @@
+import Module, {dependency} from 'parser/core/Module'
+import {Event} from 'events'
+import {Data} from 'parser/core/modules/Data'
+import {Status} from 'data/STATUSES'
+import {AbilityEvent, CastEvent} from 'fflogs'
+
+const trackedBuffEventTypes = ['applybuff', 'applybuffstack', 'removebuff', 'removebuffstack', 'refreshbuff']
+const isTrackedBuffEvent = (event: Event): event is AbilityEvent => (trackedBuffEventTypes.includes(event.type as string))
+// Statuses applied before the pull won't have an apply(de)?buff event
+// Fake buff applications so modules don't need to take it into account
+export class PrecastStatus extends Module {
+	static handle = 'precastStatus'
+	static debug = true
+
+	@dependency private data!: Data
+
+	trackedStatuses = new Map<number, number[]>()
+	trackedActions: number[] = []
+	buffEventsToSynth: AbilityEvent[] = []
+	castEventsToSynth: CastEvent[] = []
+	private startTime = this.parser.fight.start_time
+
+	normalise(events: Event[]) {
+		for (const event of events) {
+			if (event.type === 'cast') {
+				const action = this.data.getAction(event.ability.guid)
+				if (action && !this.trackedActions.includes(action.id)) {
+					this.markActionAsTracked(action.id)
+					continue
+				}
+			}
+
+			if (isTrackedBuffEvent(event)) {
+				if (event.targetID) {
+					const trackedStatusesForTarget = this.trackedStatuses.get(event.targetID)
+					if (trackedStatusesForTarget != null && trackedStatusesForTarget.includes(event.ability.guid)) {
+						// Status is already tracked for this target and no synth needs to take place
+						continue
+					}
+
+					const statusInfo = this.data.getStatus(event.ability.guid) as Status
```

Comment:
shouldn't need the cast here i don't think - the ugly af union `getStatus` returns should be a superset of `Status`, so anything that accepts `Status` should accept the union.

the union is intentional, fwiw

Reply:
I feel like I ran into some sort of TS problem without the cast, but it's not happening now so YOLO